tests/framework/lookup/test_lookup.py

```diff
@@ -253,53 +246,59 @@ def test_validate_parameters_fail_other_data(data):
 
 
 @pytest.mark.parametrize(
-    "key_cols, param_cols, val_cols",
+    "key_cols, param_cols, val_cols, match",
     [
-        (None, None, ["a", "b", "c"]),
-        (None, ["d"], ["one", "two", "three"]),
-        (["KEY"], None, ["a", "b", "c"]),
-        (["KEY"], ["d"], ["a", "b", "c"]),
+        ([], [], ["c"], "either key_columns or parameter_columns"),
+        (["a", "b"], ["b"], ["c"], "no overlap between key.*and parameter columns"),
+        (["a"], ["b"], ["a", "c"], "no overlap between value.*and key.*columns"),
+        (["a"], ["b"], ["b", "c"], "no overlap between value.*and.*parameter columns"),
+        (["d"], ["b"], ["c"], "columns.*must all be present"),
+        (["a"], ["d"], ["c"], "columns.*must all be present"),
+        (["a"], ["b"], ["d"], "columns.*must all be present"),
     ],
 )
-def test_validate_parameters_pass_scalar_data(key_cols, param_cols, val_cols):
-    validate_build_table_parameters([1, 2, 3], key_cols, param_cols, val_cols)
+def test_validate_parameters_error_dataframe(key_cols, param_cols, val_cols, match):
+    data = pd.DataFrame({"a": [1, 2], "b_start": [0, 5], "b_end": [5, 10], "c": [100, 150]})
+    with pytest.raises(ValueError, match=match):
+        validate_build_table_parameters(data, key_cols, param_cols, val_cols)
+
+
+def test_validate_parameters_pass_scalar_data():
+    validate_build_table_parameters([1, 2, 3], (), (), ["a", "b", "c"])
 
 
 @pytest.mark.parametrize(
     "key_cols, param_cols, val_cols",
-    [(["a"], ["b"], ["c"]), ([], ["b"], ["c", "a"]), ([], ["b"], ["a", "c"])],
+    [
+        (["a"], ["b"], ["c"]),
+        ([], ["b"], ["c", "a"]),
+        ([], ["b"], ["a", "c"]),
+        ([], ["b"], ["c"]),
+        (["a"], [], ["c"]),
+        (["a"], ["b"], []),
+        (["a"], [], []),
+        ([], ["b"], []),
+    ],
```

Comment:
So in the following test if `validate_build_table_parameters` raises an error the test fails and if it doesn't it then passes correct?

Reply:
That's correct!